sophia.tex

```diff
@@ -155,146 +155,187 @@ \subsubsection{Dutch auction contract}
 arrives is outside the scope of the blockchain
 
 
-\subsubsection{Oracles example}
-
-An example of the use of oracles and contracts can be illustrated with
-a little lottery example. Note that all computation on a blockchain
-should be deterministic in order to be able to validate the
-results. If not exactly the same, the corresponding state hashes will
-differ. As a consequence there is no random number generator in the
-Sophia language\footnote{Even if some kind of random function would be
-  offered, it would be deterministic and hence have a predictable
-  outcome}. (Instead, the current blockhash could be used as entropy.)
-
-Running a simple lottery in which users buy a ticket and after a while
-one draws one of the tickets as the winner, is somehow depending on
-some kind of fair randomness. If the number is known or computable at
-the start of the buying process, one might be able to figure out what
-ticket to buy to win the lottery. But if we close the lottery and then
-ask an oracle for a random number, then a trusted, but disconnected
-computation can be used to draw the winner.
-
-So assume there is such an oracle, monitoring the chain and registered
-with a reasonable query fee covering for its cost of operation. The
-oracle has an identifier, for the example say
+\subsubsection{State Channel example}
+
+Contracts can be deployed on-chain or in a state channel. State channels
+provide means for two parties for executing transactions off-chain. Once the
+channel is opened on-chain all the communication is moved off-chain and it is
+just between the two participants. It is not limited by blockchain
+throughputh, fees or gas costs. This provides a new environment to run
+contracts in. In order to make most out of it, off-chain contracts can use
+on-chain data.
+
+An example of the use of off-chain contract and an on-chain oracle can be
+illustrated with a little insurance example. Note that all computation on a
+blockchain must be deterministic in order to be able to validate the results.
+If not exactly the same, the corresponding state hashes will differ. This is
+true especially with state channels as contracts are executed by both
+participants in their own context. This requires both to share the same view
+of the chain\footnote{We achieve this by specifying the on-chain environment
+as part of the off-chain update.}.
+
+Running a simple insurance contract in which one user provides the service
+while the other insures their property for a certain timeframe from hailstorm
+requires some input from real life data. This is where oracle input comes: for
+the example we have an on-chain Oracle to produce data if there had been a
+hailstorm in a certain city. It publishes data pinned in time with a
+corresponding block height when it occurred. Later on insured participant can
```

Comment:
```suggestion
corresponding block height when it occurred. Later on insured participants can
```

Reply:
Well one is insuring the other, so only one is being insured 🤔

Reply:
Ok, then it should be "the insured participant"

Reply:
Addressed